pkg/core/xds/resource.go

```diff
@@ -7,16 +7,19 @@ import (
 	envoy_types "github.com/envoyproxy/go-control-plane/pkg/cache/types"
 	protov1 "github.com/golang/protobuf/proto"
 	"google.golang.org/protobuf/types/known/anypb"
+
+	"github.com/kumahq/kuma/pkg/plugins/policies/core/rules"
 )
 
 // ResourcePayload is a convenience type alias.
 type ResourcePayload = envoy_types.Resource
 
 // Resource represents a generic xDS resource with name and version.
 type Resource struct {
-	Name     string
-	Origin   string
-	Resource ResourcePayload
+	Name          string
+	Origin        string
+	Resource      ResourcePayload
+	ResourceOwner *rules.UniqueResourceIdentifier
```

Comment:
Maybe it's better to avoid using `Owner` in this context as we already have an ownership concept on the store level. Probably something with `Origin`, like `OriginalResource` or `ResourceOrigin` suits better here

Reply:
changed